database/src/season.rs

```diff
@@ -404,80 +173,33 @@ impl UpdateSeason {
     /// * `season_num` - Season number we'd like to update.
     ///
-    /// # Example
-    /// ```
-    /// use database::get_conn_devel as get_conn;
-    /// use database::library::{Library, InsertableLibrary, MediaType};
-    /// use database::media::{InsertableMedia, Media};
-    /// use database::tv::InsertableTVShow;
-    /// use database::season::{Season, UpdateSeason, InsertableSeason};
-    ///
-    /// let conn = get_conn().unwrap();
-    ///
-    /// let library = InsertableLibrary {
-    ///     name: "test".into(),
-    ///     location: "/dev/null".to_string(),
-    ///     media_type: MediaType::Tv,
-    /// };
-    ///
-    /// let library_id = library.insert(&conn).unwrap();
-    ///
-    /// let new_show = InsertableMedia {
-    ///     library_id: library_id,
-    ///     name: "test".into(),
-    ///     added: "test".into(),
-    ///     media_type: MediaType::Tv,
-    ///     ..Default::default()
-    /// };
-    ///
-    /// let show_id = new_show.into_static::<InsertableTVShow>(&conn).unwrap();
-    ///
-    /// let new_season = InsertableSeason {
-    ///     season_number: 1,
-    ///     added: "test".into(),
-    ///     poster: "test".into(),
-    /// };
-    ///
-    /// let season_id = new_season.insert(&conn, show_id).unwrap();
-    ///
-    /// let season = Season::get(&conn, show_id, new_season.season_number).unwrap();
-    ///
-    /// assert_eq!(season.id, season_id);
-    /// assert_eq!(season.season_number, 1);
-    /// assert_eq!(season.added, Some("test".to_string()));
-    /// assert_eq!(season.poster, Some("test".to_string()));
-    ///
-    /// let update_season = UpdateSeason {
-    ///     season_number: None,
-    ///     added: Some("test2".into()),
-    ///     poster: None,
-    ///     tvshowid: None,
-    /// };
-    ///
-    /// let _ = update_season.update(&conn, show_id, 1);
-    /// let season2 = Season::get(&conn, show_id, new_season.season_number).unwrap();
-    ///
-    /// assert_ne!(season.added, season2.added);
-    ///
-    /// Library::delete(&conn, library_id).unwrap();
     pub async fn update(
         self,
         conn: &crate::DbConnection,
-        _id: i32,
+        tv_id: i32,
         season_num: i32,
     ) -> Result<usize, DatabaseError> {
-        use crate::schema::season::dsl::*;
-        use crate::schema::tv_show;
-
-        let _ = tv_show::dsl::tv_show
-            .filter(id.eq(_id))
-            .execute_async(conn)
-            .await?;
-
-        let entry = season
-            .filter(tvshowid.eq(_id))
-            .filter(season_number.eq(season_num));
-
-        Ok(diesel::update(entry).set(self).execute_async(conn).await?)
+        let tx = conn.begin().await?;
+
+        let rows = sqlx::query!(
+            "SELECT season.* FROM season 
+            INNER JOIN tv_show WHERE tv_show.id = ? 
+            AND season.season_number = ?",
+            tv_id,
+            season_num
+        )
+        .fetch_all(conn)
```

Comment:
no need to use `fetch_all` as there wont ever be a season with the same season number for a show.